Source/WebCore/workers/service/background-fetch/BackgroundFetch.h

```diff
@@ -84,6 +84,8 @@ class BackgroundFetch : public CanMakeWeakPtr<BackgroundFetch> {
     class Record final : public BackgroundFetchRecordLoaderClient, public RefCounted<Record>, private Identified<BackgroundFetchRecordIdentifier> {
         WTF_MAKE_TZONE_ALLOCATED(Record);
     public:
+        DEFINE_VIRTUAL_REFCOUNTED;
```

Comment:
I'm confused about what this and `DECLARE_VIRTUAL_REFCOUNTED` do--just for my curiosity, would you mind explaining? Thanks!

Reply:
Then why are you using it if you don't know what it does? (cc @geoffreygaren)

You can look up this macro definition to see what it adds to your class. I'm also confused by them, which is why I personally prefer not to use them.
```

Reply:
Haha, Chris was confused too. He talked me into removing it.

It's a way to conform to the protocol used by RefPtr using virtual ref/deref instead of concreted ref/deref. Then your subclass can implement the concrete ref/deref. You need this because a concreted subclass might inherit from multiple abstract base classes, but it can have only one refcount.